simplecpp.cpp

```diff
@@ -3223,7 +3223,7 @@ void simplecpp::preprocess(simplecpp::TokenList &output, const simplecpp::TokenL
 
                 const Token * const inctok = inc2.cfront();
 
-                const bool systemheader = (inctok->op == '<');
+                const bool systemheader = (inctok->str()[0] == '<');
```

Comment:
This indicates that code code both which also checks `op` for `<` might never be executed,

Reply:
Is `inctok->op` supposed to have any meaning here? By dumping out its value every time we reach this point in a run of the `testrunner` binary, I can see it's a null character in every case except in `include7()` and `include8()`, which are both testing the case where the argument to `#include` is derived by expanding a preprocessor macro; in these cases it holds a `<` character. So perhaps it's left over from the macro expansion? (Apologies, I'm new to the `cppcheck` innards, I'm not familiar with how it works.)

By chance, until I added my new test, these were also the only cases where the filespec in `inctok->str()` used angle brackets, so for the cases tested by `testrunner` so far, testing `inctok->op` was adequate. I agree this looks wrong, but I'm unsure whether the fix should be to look at `inctok->str()` instead, or whether to fix the initialisation of `inctok->op`.

Reply:
My initial observation was based on a short look and was wrong. I am also not familiar in how it works.

`op` is set to `<` in that code above. Since with your change that is not used anymore that can be removed.

Reply:
An additional note. It seems `op` should only be set if it is a single character token. And if the token isn't already something like a name, number or comment. That wasn't done consistently though - I addressed that within #285.